src/adapters/supabase/helpers/tables/label.ts

```diff
@@ -0,0 +1,74 @@
+import { SupabaseClient } from "@supabase/supabase-js";
+
+import { Database } from "../../types/database";
+import { Super } from "./super";
+import { Context } from "../../../../types/context";
+import { WebhookEvent } from "../../../../types/github";
+
+type LabelRow = Database["public"]["Tables"]["labels"]["Row"];
+
+export class Label extends Super {
+  constructor(supabase: SupabaseClient, context: Context) {
+    super(supabase, context);
+  }
+
+  async saveLabelChange({
+    previousLabel,
+    currentLabel,
+    authorized,
+    repository,
+  }: {
+    previousLabel: string;
+    currentLabel: string;
+    authorized: boolean;
+    repository: WebhookEvent<"issues">["payload"]["repository"];
+  }): Promise<null> {
+    const { data, error } = await this.supabase.from("labels").insert({
+      label_from: previousLabel,
+      label_to: currentLabel,
+      authorized: authorized,
+      node_id: repository.node_id,
+      node_type: "Repository",
+      node_url: repository.html_url,
+    });
+
+    if (error) throw new Error(error.message);
+    return data;
+  }
+
+  async getLabelChanges(repositoryNodeId: string) {
+    const locationId = await this._getRepositoryLocationId(repositoryNodeId);
+    if (!locationId) {
+      return null;
+    }
+    return await this._getUnauthorizedLabelChanges(locationId);
+  }
+
+  async approveLabelChange(id: number): Promise<null> {
```

Comment:
are we still gonna use `/authorize`, I've never seen it used so I'm not really sure what's the use case

Reply:
We can remove it in favor of only allowing admins and billing managers to modify price related labels. 

I think this was implemented because somebody hacked the system by literally changing the label name in the settings to increase their issue pricing. 

However I think now the pricing just reads the set price label instead of recalculating it when the task is closed as complete. 

But technically this attack could happen again by the collaborator changing the price label in the settings and making it larger just before the issue is closed as complete (which might be more obvious? Not sure)

In conclusion I vaguely recall this being implemented in a rush and yes I agree we never actually used it in production. Perhaps it's not a fully thought through feature. 

But now you know the lore.